show-callout/src/main/java/com/esri/arcgismaps/sample/showcallout/components/ComposeMapView.kt

```diff
@@ -0,0 +1,101 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.showcallout.components
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import com.arcgismaps.mapping.view.MapView
+import com.arcgismaps.mapping.view.SingleTapConfirmedEvent
+import kotlinx.coroutines.launch
+
+/**
+ * Wraps the MapView in a Composable function.
+ */
+@Composable
+fun ComposeMapView(
+    modifier: Modifier = Modifier,
+    mapViewModel: MapViewModel,
+    onSingleTap: (SingleTapConfirmedEvent) -> Unit = {}
+) {
+    // get an instance of the current lifecycle owner
+    val lifecycleOwner = LocalLifecycleOwner.current
+    // collect the latest state of the MapViewState
+    val mapViewState by mapViewModel.mapViewState.collectAsState()
+    // create and add MapView to the activity lifecycle
+    val mapView = createMapViewInstance(lifecycleOwner).apply {
+        map = mapViewState.arcGISMap
+        setViewpoint(mapViewState.viewpoint)
+        // enable animated callout
+        callout.isAnimationEnabled = true
+    }
+
+    // wrap the MapView as an AndroidView
+    AndroidView(
+        modifier = modifier,
+        factory = { mapView },
+        // recomposes the MapView on changes in the MapViewState
+        update = { mapView ->
+            mapView.apply {
+                val latlonPoint = mapViewModel.latLonPoint
+                latlonPoint?.let {
+                    // show callout at the tapped location using the set View
+                    callout.show(
+                        mapViewModel.calloutContent,
+                        latlonPoint
+                    )
+                    lifecycleOwner.lifecycleScope.launch {
+                        // center the map on the tapped location
+                        setViewpointCenter(latlonPoint)
+                    }
+                }
+            }
+        }
+    )
+
+    // launch coroutine functions in the composition's CoroutineContext
+    LaunchedEffect(Unit) {
+        mapView.onSingleTapConfirmed.collect {
+            onSingleTap(it)
```

Comment:
After this Friday's coffee meeting(9/29), I think you need to change this line to
```
launch {
                onSingleTap(it)
            }
```

Reply:
It is OK for this case, `onSingleTap` doesn't block anything